native/packages/react-native-bpk-component-animate-height/src/BpkAnimateHeight.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  View,
+  Platform,
+  Animated,
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const tokens = Platform.select({
+  ios: () => require('bpk-tokens/tokens/ios/base.react.native.common.js'), // eslint-disable-line global-require
+  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
+})();
+
+const {
+  animationDurationBase,
+} = tokens;
+
+class BpkAnimateHeight extends React.Component {
+
+  state = {
+    expanded: null,
+    expandedHeight: null,
+    collapsedHeight: 0,
+    height: new Animated.Value(1),
+    heightSet: false,
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.expanded !== this.state.expanded) {
+      this.state.expanded = nextProps.expanded;
+
+      this.resize();
+    }
+  }
+
+  setExpandedHeight = (event, expandedImmediately) => {
+    // if (this.state.expandedHeightSet) return;
+    const { height } = event.nativeEvent.layout;
+    this.setState({
+      // if the component has started expanded, we should now set the height of the container
+      // why on earth does a value of 1 instead of 0 make this break!
+      height: new Animated.Value(expandedImmediately ? height : 0),
+      heightSet: true,
+      expandedHeight: height,
+    });
+  };
+
+  resize=() => {
+    const finalValue = this.state.expanded ? this.state.expandedHeight : this.state.collapsedHeight;
+
+    Animated.timing(
+      this.state.height,
+      {
+        toValue: finalValue,
+        duration: animationDurationBase,
+      },
+      ).start();
+  }
+
+  render() {
+    const { expanded, style, ...rest } = this.props;
+
+    const { height } = this.state;
+
+    return (
+      <Animated.View
+        style={{
+          ...style,
+          overflow: 'hidden',
+          height,
+        }}
+        {...rest}
+      >
+        <View
+          onLayout={event => (this.state.heightSet ? undefined : this.setExpandedHeight(event, expanded))}
+        >
+          {this.props.children}
+        </View>
+      </Animated.View>
+    );
+  }
+}
+
+BpkAnimateHeight.propTypes = {
+  expanded: PropTypes.bool.isRequired,
+  children: PropTypes.oneOfType([
```

Comment:
why not something like `children: PropTypes.node.isRequired,`? are you expecting an array of children?

Reply:
It complained when I did that!